mantidimaging/core/utility/cuda_check.py

```diff
@@ -0,0 +1,27 @@
+# Copyright (C) 2020 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+
+import os
+from typing import Tuple
+
+
+def _read_from_terminal(command: str) -> str:
+    """
+    Runs a terminal command and returns the result.
+    """
+    return os.popen(command).read()
+
+
+def cuda_is_present() -> bool:
+    """
+    Checks if nvidia-smi is on the system + working, and that the libcuda file can be located.
+    """
+    return "Driver Version" in _read_from_terminal("nvidia-smi") and _read_from_terminal("locate libcuda.so") != ""
```

Comment:
I'm not sure that `_read_from_terminal("locate libcuda.so")` will be very accurate - `locate` seems to be returning `libcuda.so`s for me that have been installed in software (Avizo) or the manpages entry:

```
/home/dtasev/dev/containers/avizo/avizo/usr/local/cuda-10.0/compat/libcuda.so
/home/dtasev/dev/containers/avizo/avizo/usr/local/cuda-10.0/compat/libcuda.so.1
/home/dtasev/dev/containers/avizo/avizo/usr/local/cuda-10.0/compat/libcuda.so.410.104
/home/dtasev/dev/containers/mantidimaging/mi_savu.sif/usr/local/cuda-9.0/targets/x86_64-linux/lib/stubs/libcuda.so
/usr/lib/i386-linux-gnu/libcuda.so
/usr/lib/i386-linux-gnu/libcuda.so.1
/usr/lib/i386-linux-gnu/libcuda.so.440.33.01
/usr/lib/x86_64-linux-gnu/libcuda.so
/usr/lib/x86_64-linux-gnu/libcuda.so.1
/usr/lib/x86_64-linux-gnu/libcuda.so.440.33.01
/usr/local/cuda-10.2/doc/man/man7/libcuda.so.7
/usr/local/cuda-10.2/targets/x86_64-linux/lib/stubs/libcuda.so
/usr/share/man/man7/libcuda.so.7
```

Is there any way we can filter this to just be from the system libraries (note that they're not the same between distributions, `/usr/lib/x86_64-linux-gnu/` for x64 debian (ubuntu), and `/usr/lib64` for rhel/centos iirc. Best way would be if we can call `locate` with a flag that limits it to system dirs, but I'm not sure one exists.

Any other suggestions? @Pasarus @samtygier-stfc